tests/test_build_nanoset_dataloader.py

```diff
@@ -0,0 +1,130 @@
+import random
+
+import pytest
+from helpers.context import TestContext
+from helpers.data import (
+    assert_batch_dataloader,
+    create_dataset_paths,
+    create_dummy_json_dataset,
+    get_max_value_by_group,
+    preprocess_dummy_dataset,
+)
+from helpers.utils import available_gpus, get_all_3d_configurations, init_distributed, rerun_if_address_is_in_use
+from nanotron.data.blended_nanoset import BlendedNanoset
+from nanotron.data.dataloader_builder import build_nanoset_dataloader
+from nanotron.data.dataset_builder import NanosetBuilder
+from nanotron.data.nanoset import Nanoset
+from nanotron.data.nanoset_configs import NanosetConfig
+from nanotron.data.utils import compute_datasets_num_samples
+from nanotron.parallel import ParallelContext
+
+
+@pytest.mark.parametrize(
+    "tp,dp,pp",
+    [
+        pytest.param(*all_3d_configs)
+        for gpus in range(1, min(available_gpus(), 4) + 1)
+        for all_3d_configs in get_all_3d_configurations(gpus)
+    ],
+)
+@rerun_if_address_is_in_use()
+def test_build_nanoset_dataloader(tp: int, dp: int, pp: int):
+    test_context = TestContext()
+
+    # Create dataset files
+    json_paths, bin_paths = create_dataset_paths(tmp_dir=test_context.get_auto_remove_tmp_dir(), quantity=2)
+
+    # Create dummy json datasets
+    for idx, json_path in enumerate(json_paths):
+        create_dummy_json_dataset(path_to_json=json_path, dummy_text=f"Nanoset {idx}!")
+
+    # Preprocess dummy json datasets
+    for json_path in json_paths:
+        preprocess_dummy_dataset(path_to_json=json_path)
+
+    init_distributed(tp=tp, dp=dp, pp=pp)(_test_build_nanoset_dataloader)(
+        test_context=test_context, path_to_bin_files=bin_paths
+    )
+
+
+def _test_build_nanoset_dataloader(
+    parallel_context: ParallelContext, test_context: TestContext, path_to_bin_files: str
+):
+    SEED = 1234
+    random.seed(SEED)
+
+    TRAIN_STEPS = random.randint(1000, 10000)
```

Comment:
For easier to detect what cause a test fail later on, maybe we should fix the train_steps? Parameterize it [100, 500] or something